src/authorization/client/enterprise.rs

```diff
@@ -0,0 +1,45 @@
+use async_trait::async_trait;
+use o2_enterprise::enterprise::openfga::authorizer::OpenFgaConfig;
+
+use super::{AuthorizationClientTrait, ObjectType};
+use crate::common::meta::authz::Authz;
+
+/// Client for performing authorization checks using OpenFGA.
+#[derive(Debug, Clone)]
+pub struct AuthorizationClient {
+    pub open_fga_config: OpenFgaConfig,
+    pub open_fga_store_id: String,
+    pub page_size: i32,
+}
+
+#[cfg(feature = "enterprise")]
+impl AuthorizationClient {
+    pub fn new() -> Self {
+        todo!()
+    }
```

Comment:
TODO: Pass in the configuration after it is read from environment variables in the main function